room_allocation.py

```diff
@@ -104,22 +122,248 @@ def get_room_by_name(self, room_name):
 
+    def get_person_room(self, person_name):
+        for room in self.rooms.values():
+            if person_name in room.people_in_room:
+                return room
+
+    def reallocate_person(self, person_name, room_name):
+        '''
+        Move person from the current room to a specified room
+        args:
+            person_name - name of person to be moved
+            room_name - name of room to move person to
+        '''
+        if person_name not in self.all_people:
+            return "Person does not exist."
+        if room_name not in self.rooms:
+            return "Room does not exist."
+
+        current_room = self.get_person_room(person_name)
+        person = self.all_people[person_name]
+        new_room = self.rooms[room_name]
+
+        if self.can_be_in_room(person, new_room) and current_room:
+            current_room.people_in_room.pop(person_name)
+            new_room.people_in_room[person_name] = person
+            if person_name in self.unallocated_people:
+                self.unallocated_people.pop(person_name)
+            return "Moved person from %s to %s." % (current_room.name, new_room.name)
+        elif not current_room:  # person is in unallocated
+            new_room.people_in_room[person_name] = person
+            self.unallocated_people.pop(person_name)
+            return "Moved person to room %s." % (new_room.name)
+        else:
+            return "Person cannot be in room."
+
+    def load_people(self, filepath):
+        '''
+        Loads people into amity from file
+        args:
+            filepath - directory of file to load people from
+        '''
+        if not os.path.isfile(filepath):
+            return "Invalid file path."
+
+        f = open(filepath, "r")
+        data = f.read().upper()
+        f.close()
+        if len(data) < 1:
+            return "File is empty."
+
+        data = data.split('\n')
+
+        for person_data in data:
+            person_data = person_data.split(' ')
+            person_name = person_data[0] + ' ' + person_data[1]
+
+            if 'FELLOW' in person_data:
+                switch = {'Y': True, 'N': False}
+                try:
+                    wants_accommodation = switch[person_data[3]]
+                except KeyError:
+                    return "File data was written in the wrong format."
+
+                person = self.add_person(
+                    person_name, 'fellow',
+                    wants_accommodation=wants_accommodation)
+
+            elif 'STAFF' in person_data:
+                person = self.add_person(person_name, 'staff')
+
+        return "Load was successful."
+
+    def print_allocations(self, filename=None):
+        '''
+        Prints out each room's name and the people in it;
+        this can be piped into a file if specified in the filename parameter.
+        Overwrites if piping.
+        '''
+        if len(self.rooms) < 1:
+            return "No rooms exist."
```

Comment:
You might have unallocated people. You can also print them in print_allocations if there are no rooms